AxonDeepSeg/segment.py

```diff
@@ -247,6 +255,13 @@ def main(argv=None):
 
     # Processing the arguments
     args = vars(ap.parse_args(argv))
+
+    if "version" in args:
+        print(f"AxonDeepSeg v.{AxonDeepSeg.__version__} ({git.Repo(search_parent_directories=True).head.object.hexsha})")
+        sys.exit(0)
+
+    logger.info(f"AxonDeepSeg v.{AxonDeepSeg.__version__} ({git.Repo(search_parent_directories=True).head.object.hexsha})")
```

Comment:
It might be good to make this string a constant instead? Maybe inside `__init__.py`, since we already define `__version__` there?

Something like:

```
import git

__version__ = "5.0.0"
__git_version__ = git.Repo(search_parent_directories=True).head.object.hexsha
__version_string__ = f"AxonDeepSeg v{__version__} ({__git_version__})"
```

That way, it's easy to access `AxonDeepSeg.__version_string__` not only in `segment.py`, but anywhere else we may want it too.

Reply:
Also, tiny nitpick, but how do you feel about `v{__version__}` (i.e. `v5.0.0` rather than `v.5.0.0`)?

Reply:
good point for both! 

I'm also fine with the second re point, I don't recall any reason we set it that way in the first place but I agree the second appears better and is more consistent with othter softwares styles

Reply:
@joshuacwnewton all done! Here's the outputs:

(venv_ads) mathieuboudreau@arm64-apple-darwin20 axondeepseg % axondeepseg -V
AxonDeepSeg v5.0.0 (3bddcb6eaa201da86e43074500bb3b4eae24f787)
(venv_ads) mathieuboudreau@arm64-apple-darwin20 axondeepseg % axondeepseg --version
AxonDeepSeg v5.0.0 (3bddcb6eaa201da86e43074500bb3b4eae24f787)
(venv_ads) mathieuboudreau@arm64-apple-darwin20 axondeepseg % axondeepseg -i test/__test_files__/__test_demo_files__/image.png
2024-12-13 13:27:18.198 | INFO     | AxonDeepSeg.segment:main:260 - AxonDeepSeg v5.0.0 (3bddcb6eaa201da86e43074500bb3b4eae24f787)
perform_everything_on_gpu=True is only supported for cuda devices! Setting this to False
2024-12-13 13:27:18.214 | INFO     | AxonDeepSeg.apply_model:axon_segmentation:148 - Running inference on device: cpu

Reply:
Looks great! Though, I notice that the `logger.info` call could be updated to use  `AxonDeepSeg.__version_string__` as well?

Reply:
Done!